### PR TITLE
Shape opt opt utils

### PR DESCRIPTION
**Pull request by adityaghantasala**
**Description**
This is the first of a series of changes that will follow to make it possible that shape optimization application can deal with multiple design surfaces with different VertexMorphing parameters. 

**Changelog**
Modifies the optimization_utilities.h so no design surface and parameters are required in the constructor. 
Changes necessary files to make the above change possible. 

All the tests in ShapeOptApp pass locally.

### CODE REVIEW

**Review by armingeiser — COMMENTED** [review]
Looks good for me in general.

One question: Do you plan that the individual design surfaces are "connected", i mean do they share a common edge, or are will they be geometrically separated?

**Comment by adityaghantasala** [comment]
> Looks good for me in general.
> 
> One question: Do you plan that the individual design surfaces are "connected", i mean do they share a common edge, or are will they be geometrically separated?

The first version I will do is considering geometrically separated. If you think it will be easy if we consider "connected" design surfaces from the beginning, we can discuss this.

**Comment by armingeiser** [comment]
> If you think it will be easy if we consider "connected" design surfaces from the beginning, we can discuss this.

Connected design surfaces require some special attention for the intersection to remain smooth. So i would say starting with separated design surfaces is good for the moment.

**Review by armingeiser — DISMISSED** [review]

**Comment by adityaghantasala** [comment]
@armingeiser Thank you. I will wait more to see if someone else has any comments and then merge !

**Comment by sunethwarna** [comment]
Hi, As I see, all the methods are standalone ryt (no member variable dependence)? so would it be possible to make all of them static?

**Comment by adityaghantasala** [comment]
> Hi, As I see, all the methods are standalone ryt (no member variable dependence)? so would it be possible to make all of them static?

Done@suneth

File: applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 1]
```diff
@@ -72,28 +72,28 @@ inline void InverseMapScalar(TMapper& mapper,
     mapper.InverseMap(origin_variable, destination_variable);
 }
 
-inline double ComputeL2NormScalar(OptimizationUtilities& utils, const Variable< double >& variable)
+inline double ComputeL2NormScalar(ModelPart& rModelPart, const Variable< double >& variable)
```
Comment (sunethwarna) [inline applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp]: I was thinking about making those also static? Would it be too much work?

File: applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 2]
```diff
@@ -163,30 +163,29 @@ void  AddCustomUtilitiesToPython(pybind11::module& m)
     // For performing individual steps of an optimization algorithm
     // ========================================================================
     py::class_<OptimizationUtilities >(m, "OptimizationUtilities")
-        .def(py::init<ModelPart&, Parameters>())
+        .def(py::init<>())
```
Comment (sunethwarna) [inline applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp]: If we can make the c++ methods static, then we can remove this constructor too.

Comment (adityaghantasala) [reply applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp]: done

File: applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp (hunk 1 above)
Comment (adityaghantasala) [reply applications/ShapeOptimizationApplication/custom_python/add_custom_utilities_to_python.cpp]: moved to lambdas now

**Review by sunethwarna — APPROVED** [review]
@adityaghantasala thanks :)

**Pull request merged**